src/migrations/migrate_sync_visible_flag.js

```diff
@@ -0,0 +1,16 @@
+import { isLocalStorageAvailable } from '../util/settings_persister';
+
+// Fixes a typo in the config: `visibile` !== `visible`
+export default () => {
+  if (!isLocalStorageAvailable) {
+    return;
+  }
+
+  const shouldSyncOnBecomingVisible = localStorage.getItem('shouldSyncOnBecomingVisibile');
+  if (!shouldSyncOnBecomingVisible) {
+    return;
+  }
+
+  localStorage.setItem('shouldSyncOnBecomingVisibile', shouldSyncOnBecomingVisible);
```

Comment:
Is it just me or is this another typo?  Shouldn't it be this instead?

```suggestion
  localStorage.setItem('shouldSyncOnBecomingVisible', shouldSyncOnBecomingVisible);
```

That sounds like it would explain why the migration is not working for you, which *might* in turn imply that the problem here is *not* related to the problems in #320 after all.

Reply:
Good catch :pray:, but unfortunately no. I made a screencast where you can see that the current configuration shows in the settings page and in the JSON. Then, when switching to the new branch (with the typo fixed), the setting doesn't show anymore and is removed without replacement in the config JSON.

![should_sync](https://user-images.githubusercontent.com/505721/84185552-79e02780-aa8f-11ea-8828-d0539e37b2a2.gif)